nixos/modules/misc/mandoc.nix

```diff
@@ -0,0 +1,98 @@
+{ config, lib, pkgs, ... }:
+
+let
+  makewhatis = "${lib.getBin pkgs.mandoc}/bin/makewhatis";
+
+  docCfg = config.documentation;
+  cfg = docCfg.mandoc;
+
+in {
+  meta.maintainers = [ lib.maintainers.sternenseemann ];
+
+  options = {
+    documentation.mandoc = {
+      enable = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Whether to enable installation and viewing
+          of man pages using the utilities from
+          <literal>mandoc</literal>.
+          This can be used as a drop-in replacement
+          for <option>documentation.man</option>.
+          If you also want to additionally install the
+          man pages from the <literal>devman</literal>
+          output of installed packages, set the
+          <option>documentation.dev.enable</option>
+          to <literal>true</literal>.
+        '';
+      };
+
+      manPath = lib.mkOption {
+        type = with lib.types; listOf str;
+        default = [ "share/man" ];
+        example = lib.literalExample "[ \"share/man\" \"share/man/fr\" ]";
+        description = ''
+          Change the manpath, i. e. the directories where
+          <citerefentry><refentrytitle>man</refentrytitle><manvolnum>1</manvolnum></citerefentry>
+          looks for section-specific directories of man pages.
+          You only need to change this setting if you want extra man pages
+          (e. g. in non-english languages). All values must be strings that
+          are a valid path from the target prefix (without including it).
+          The first value given takes priority.
+        '';
+      };
+
+      generateCaches = lib.mkOption {
+        type = lib.types.bool;
+        default = false;
+        description = ''
+          Whether to generate a <literal>mandoc.db</literal> indexing
+          the installed man pages for
+          <citerefentry><refentrytitle>apropos</refentrytitle><manvolnum>1</manvolnum></citerefentry>,
+          <citerefentry><refentrytitle>whatis</refentrytitle><manvolnum>1</manvolnum></citerefentry>
+          and the <literal>-k</literal> option of
+          <citerefentry><refentrytitle>man</refentrytitle><manvolnum>1</manvolnum></citerefentry>
+          using
+          <citerefentry><refentrytitle>makewhatis</refentrytitle><manvolnum>8</manvolnum></citerefentry>.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    assertions = [
+      {
+        assertion = !(cfg.enable && docCfg.man.enable);
+        message = ''
+          documentation.mandoc is a replacement for documentation.man,
+          so they can't both be enabled at the same time.
+        '';
+      }
+    ];
+
+    environment = {
+      systemPackages = [ pkgs.mandoc ];
```

Comment:
I'd like to see this become a `package` option, especially since `pkgs.mandoc` is referred to in a very hard to override way elsewhere in the module expression (`makewhatis = "${lib.getBin pkgs.mandoc}/bin/makewhatis";`).

`documentation.man.man-db.package` should probably become a thing too.